### PR TITLE
Only precompile Pkg.jl when generating output

### LINKED ISSUE #3691: Loading Pkg with --compiled-modules=no changes active project
Starting Julia (nightly) with `--compiled-modules=no` and then loading Pkg changes the currently active environment:
```
$ julia --compiled-modules=no --project=. -e 'println(Base.active_project()); using Pkg; println(Base.active_project())'
/home/pfitzseb/Documents/Git/julia-vscode/scripts/packages/SymbolServer/Project.toml
/tmp/jl_Cs4MSb/TestPkg.jl/Project.toml

$ julia -e 'using InteractiveUtils; versioninfo()'
Julia Version 1.11.0-DEV.874
Commit 137783f166 (2023-11-09 04:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: 8 × Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  WORD_SIZE: 64
  LLVM: libLLVM-15.0.7 (ORCJIT, skylake)
  Threads: 1 on 8 virtual cores
```

@fredrikekre pointed out that [this](https://github.com/JuliaLang/Pkg.jl/blob/2d59169bffdf29a4ed7bafd8b35c391ede612dde/src/precompile.jl#L116-L118) is most likely to blame.

cc @mkitti

### PR DESCRIPTION
**Pull request by mkitti**
Fix #3691 for real

We probably should also create a test

### CODE REVIEW

**Comment by IanButterworth** [comment]
It would be a lot clearer to just add an if, and not mess with the indenting (which is odd in this file)
```
if Base.generating_output()
    pkg_precompile()
end
```

**Comment by IanButterworth** [comment]
Please also edit PRs, not open another for changes.

**Comment by mkitti** [comment]
Sorry, I've been trying to do this on mobile between other activities. I will be at a computer shortly.

**Comment by mkitti** [comment]
The diff has been minimized

**Comment by mkitti** [comment]
@IanButterworth here is the 3 line patch

File: src/precompile.jl [hunk 1]
```diff
@@ -138,5 +138,7 @@ function pkg_precompile()
     return nothing
 end
 
-pkg_precompile()
+    if Base.generating_output()
+        pkg_precompile()
+    end
```
Comment (IanButterworth) [inline src/precompile.jl:143]:
```suggestion
if Base.generating_output()
    pkg_precompile()
end
```

**Pull request merged**